### PR TITLE
Globular types

### PR DESCRIPTION
**Pull request by fredrik-bakke**
Defines small and large globular types.

### CODE REVIEW

File: src/structured-types/globular-types.lagda.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+### The structure of a globular type
+
+```agda
+record globular-structure {l : Level} (A : UU l) : UU (lsuc l)
+  where
+  coinductive
+  field
+    1-cell-globular-structure :
+      (x y : A) → UU l
+    higher-globular-structure :
```
Comment (fredrik-bakke) [inline src/structured-types/globular-types.lagda.md:64]: It seems like a name for this field that fits into a general scheme for higher cells is `globular-structure-1-cell-globular-structure`

**Comment by fredrik-bakke** [comment]
While working on this PR I started wondering if we should define `Large-Type` as 
```agda
Large-Type : (Level → Level) → UUω
Large-Type α = (l : Level) → UU (α l)
```
and that `UUω` should rather be `Typeω`.

So far, we usually take a large thing to be a hierarchy of small things, not to be a literal thing in `UUω`. E.g. `Large-Precategory` and so on.

**Comment by fredrik-bakke** [comment]
Turns out the terminology "reflexive globular type" is not unprecedented: https://ncatlab.org/nlab/show/globular+set#ReflexiveGlobularSets

File: src/structured-types/globular-types.lagda.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  globular-type-2-cell-Globular-Type f g =
+    ( 2-cell-Globular-Type f g , globular-structure-2-cell-Globular-Type f g)
+```
+
+## Examples
+
+### The globular structure on a type given by its identity types
+
+```agda
+globular-structure-Id : {l : Level} (A : UU l) → globular-structure A
```
Comment (VojtechStep) [inline src/structured-types/globular-types.lagda.md:177]: What do you think about the definition using `λ where`, which avoids repeating the name and arguments? To me it reads better:
```agda
globular-structure-Id A = λ where
  .1-cell-globular-structure x y → x ＝ y
  .globular-structure-1-cell-globular-structure x y → globular-structure-Id (x ＝ y)
```

File: src/structured-types/infinity-magmoids.lagda.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+open import foundation.universe-levels
+
+open import structured-types.globular-types
+```
+
+</details>
+
+## Idea
+
+An {{#concept "$∞$-magmoid" Agda=∞-Magmoid}} is a
```
Comment (VojtechStep) [inline src/structured-types/infinity-magmoids.lagda.md:23]: Does `$∞$` render differently than `∞`?

File: src/structured-types/large-globular-types.lagda.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+  {A : (l : Level) → UU (α l)}
+  (G : large-globular-structure β A)
+  {l1 l2 : Level} {x : A l1} {y : A l2}
+  (f g : 1-cell-large-globular-structure G x y)
+  where
+
+  2-cell-large-globular-structure : UU (β l1 l2)
+  2-cell-large-globular-structure =
+    1-cell-globular-structure
+      ( globular-structure-1-cell-large-globular-structure G x y) f g
```
Comment (VojtechStep) [inline src/structured-types/large-globular-types.lagda.md:85]:
```suggestion
      ( globular-structure-1-cell-large-globular-structure G x y)
      ( f)
      ( g)
```
... and so on in the file

File: src/structured-types/globular-types.lagda.md (hunk 2 above)
Comment (VojtechStep) [reply src/structured-types/globular-types.lagda.md:177]: Oh, now I see that you're using it for the other definitions in latter files 👌

File: src/structured-types/symmetric-globular-types.lagda.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+### The symmetric globular structure on a type given by its identity types
+
+```agda
+is-symmetric-globular-structure-Id :
+  {l : Level} (A : UU l) →
+  is-symmetric-globular-structure (globular-structure-Id A)
+is-symmetric-globular-structure-Id A =
+  λ where
+  .is-symmetric-1-cell-is-symmetric-globular-structure x y → inv
```
Comment (VojtechStep) [inline src/structured-types/symmetric-globular-types.lagda.md:92]:
```suggestion
  .is-symmetric-1-cell-is-symmetric-globular-structure x y →
    inv
```
to be consistent with how we split multi-part definitions where one of the components doesn't fit on one line; alternatively keep this as it is and change reflexivity structure on the identity type in reflexive-globular-types.

**Review by VojtechStep — COMMENTED** [review]
Looks great 👍

File: src/structured-types/infinity-magmoids.lagda.md (hunk 3 above)
Comment (fredrik-bakke) [reply src/structured-types/infinity-magmoids.lagda.md:23]: Very much so.

image

**Pull request merged**